crates/wdl-engine/src/backend/lsf_apptainer.rs

```diff
@@ -0,0 +1,648 @@
+        // resource request. That would likely mean using `bsub -n min,max`
+        // syntax as it doesn't seem that `affinity` strings support ranges
+        let _max_cpu = v1::max_cpu(hints);
+        // TODO ACF 2025-09-11: set a hard memory limit with `bsub -M !`?
+        let _max_memory = v1::max_memory(hints)?.map(|i| i as u64);
+
+        // Truncate the request ID to fit in the LSF job name length limit.
+        let request_id = request.id();
+        let name = if request_id.len() > LSF_JOB_NAME_MAX_LENGTH {
+            request_id
+                .chars()
+                .take(LSF_JOB_NAME_MAX_LENGTH)
+                .collect::<String>()
+        } else {
+            request_id.to_string()
+        };
+
+        self.manager.send(
+            LsfApptainerTaskRequest {
+                backend_config: self.backend_config.clone(),
+                spawn_request: request,
+                name,
+                container,
+                cpu,
+                memory,
+                crankshaft_events: self.crankshaft_events.clone(),
+                cancellation_token,
+            },
+            completed_tx,
+        );
+
+        Ok(completed_rx)
+    }
+
+    fn cleanup<'a>(
+        &'a self,
+        _work_dir: &'a EvaluationPath,
+        _token: CancellationToken,
+    ) -> Option<futures::future::BoxFuture<'a, ()>> {
+        // TODO ACF 2025-09-11: determine whether we need cleanup logic here;
+        // Apptainer's security model is fairly different from Docker so
+        // uid/gids on files shouldn't be as much of an issue, and using only
+        // `apptainer exec` means no longer-running containers to tear down
+        None
+    }
+}
+
+/// Configuration for the LSF + Apptainer backend.
+// TODO ACF 2025-09-12: add queue option for short tasks
+//
+// TODO ACF 2025-09-23: add a Apptainer/Singularity mode config that switches around executable
+// name, env var names, etc.
+#[derive(Debug, Clone, serde::Deserialize, serde::Serialize)]
+pub struct LsfApptainerBackendConfig {
+    /// Which queue, if any, to specify when submitting jobs to LSF.
+    pub queue: Option<String>,
+    /// The maximum number of scatter subtasks that can be evaluated
+    /// concurrently.
+    ///
+    /// By default, this is 200.
+    #[serde(default = "default_max_scatter_concurrency")]
+    pub max_scatter_concurrency: u64,
+    /// The directory in which temporary directories will be created containing
+    /// Apptainer `.sif` files.
+    ///
+    /// This should be a location that is accessible by all jobs on the LSF
+    /// cluster.
+    ///
+    /// By default, this is `~/.cache/sprocket-apptainer-images`.
+    #[serde(default = "default_apptainer_images_dir")]
+    pub apptainer_images_dir: PathBuf,
+}
+
+fn default_max_scatter_concurrency() -> u64 {
+    200
+}
+
+fn default_apptainer_images_dir() -> PathBuf {
+    PathBuf::from("~/.cache/sprocket-apptainer-images")
+}
```

Comment:
Should this do something similar to [config](https://github.com/stjude-rust-labs/sprocket/blob/ddd42c3fff176a95f08f45c89889a35443ce8ce1/src/config.rs#L145-L150) and use `XDG_CACHE_HOME` rather than always the user's home directory?

Reply:
Ah, I pushed a fix that manually tacks `.cache` on, but it's nice to respect the proper env var. I'll push another shortly.